agents/agents-features/agents-features-memory/src/commonMain/kotlin/ai/koog/agents/memory/feature/nodes/MemoryNodes.kt

```diff
@@ -153,6 +167,11 @@ public fun <T> AIAgentSubgraphBuilderBase<*, *>.nodeSaveToMemoryAutoDetectFacts(
                 }
             }
         }
+
+        dropLastNMessages(2) // remove the prompt and LLM response
```

Comment:
Maybe the way to go here is make a deep copy of the prompt, then run fact retrieving and then replace back? Constant looks friteting here (and in case something will change inside requestLLMWithoutTools we will have a hidden bug)